src/eos-shard-writer.c

```diff
@@ -19,10 +19,19 @@
 
 #include "eos-shard-writer.h"
 
-#include <endian.h>
 #include <fcntl.h>
 #include <string.h>
 
+#include "config.h"
+#ifdef HAVE_ENDIAN_H
+# include <endian.h>
+#elif defined(HAVE_LIBKERN_OSBYTEORDER_H)
+# include <libkern/OSByteOrder.h>
+# define htole64(x) OSSwapHostToLittleInt64(x)
```

Comment:
> doesn't that mean your check for endian.h doesn't quite work?

OSX at least had the foresight to name it `machine/endian.h` so that doesn't conflict. The check does work.

> i'm half-tempted to use something like https://gist.github.com/panzi/6856583

That was my reference for the OSX byte swapping code :-)

> I think GUINT64_TO_LE should work. Can you submit a patch that just uses that everywhere?

I'll take a look tomorrow.